internal/crypto/cms/signed.go

```diff
@@ -0,0 +1,216 @@
+package cms
+
+import (
+	"bytes"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"time"
+
+	"github.com/notaryproject/notation-go-lib/internal/crypto/hashutil"
+	"github.com/notaryproject/notation-go-lib/internal/crypto/oid"
+)
+
+// ParsedSignedData is a parsed SignedData structure for golang friendly types.
+type ParsedSignedData struct {
+	Content      []byte
+	ContentType  asn1.ObjectIdentifier
+	Certificates []*x509.Certificate
+	CRLs         []pkix.CertificateList
+	Signers      []SignerInfo
+}
+
+// ParseSignedData parses ASN.1 DER-encoded SignedData structure to golang friendly types.
+func ParseSignedData(data []byte) (*ParsedSignedData, error) {
+	var contentInfo ContentInfo
+	if _, err := asn1.Unmarshal(data, &contentInfo); err != nil {
+		return nil, SyntaxError{Message: "invalid content info", Detail: err}
+	}
+	if !oid.SignedData.Equal(contentInfo.ContentType) {
+		return nil, ErrExpectSignedData
+	}
+
+	var signedData SignedData
+	if _, err := asn1.Unmarshal(contentInfo.Content.Bytes, &signedData); err != nil {
```

Comment:
We should add validation for `SignedData#Version`

Reply:
Accroding to [RFC 5652 5.1](https://datatracker.ietf.org/doc/html/rfc5652#section-5.1), the `SignedData` struct can handle all versions (1,3,4,5). I don't think we need to valid the versions unless we have strong security reasons.